session.go

```diff
@@ -407,7 +413,22 @@ func (s *Session) handleAckFrame(frame *frames.AckFrame) error {
 }
 
 // Close the connection. If err is nil it will be set to qerr.PeerGoingAway.
+// It waits until the run loop has stopped before returning
 func (s *Session) Close(e error) error {
+	err := s.closeImpl(e, false)
+
+	if atomic.LoadUint32(&s.closed) == 1 {
```

Comment:
You could test for that using `Expect(s.runClosed).To(NotReceive())` or similar, since the channel should have been drained here.